Enable increment_usage() to increment by more than 1 at a time

jamesmishra opened the pull request:
This helps with various rate limiting schemes where different
operations have different costs or credit usages.
increment_usage() can be used to increment the rate limiter
by an arbitrary amount.

This code also does not screen for negative numbers, so it
is possible to decrement the rate limit count if necessary.

italorossi commented:
Hello there, test are broken and please make sure to use TDD to the new behavior being added.

jamesmishra commented:
Thanks for the fast reply! I looked over Travis CI errors and made a few changes:

 - I reverted all changes to `get_wait_time` because they would change behavior for existing users of this package.
 - I reverted the error messages in exceptions to the way they were before. Tests were failing because I had used Python 3.6 f-strings to make the messages. We can also choose to bring those messages back with Python 2-compatible string formatting.

In the next few days, I'll add tests for the changed functionality.

victor-torres commented on redis_rate_limit/__init__.py:
```diff
@@ -103,21 +113,27 @@ def has_been_reached(self):
         :return: integer: current usage
         """
+        if increment_by > self._max_requests:
+            raise InitialValueTooHigh()
```
You could just use a generic ValueError exception with a good message such as:

> The increment_by parameter should not be greater than the maximum number of requests.

Also, consider adding a blank line after the if-statement.

italorossi replied:
if increment_by > self._max_requests or increment_by < 0:
    raise ValueError

jamesmishra replied:
@italorossi Do we want to allow "incrementing" by 0? As the code is right now, it would evaluate the Redis script, but nothing would change.

victor-torres commented on redis_rate_limit/__init__.py:
```diff
@@ -103,21 +113,27 @@ def has_been_reached(self):
         method raises an Exception.
 
+        :param increment_by: The count to increment the rate limiter by.
+        This is typically 1, but higher or lower values are provided
```
This sounds a little bit confusing to me. So, will it be possible to **increment** the rate limiter with **negative** values?

That's not making much sense to me. Could you clarify with an use case?

jamesmishra replied:
Yeah, the code doesn't currently check against negative values, but I could add one.

I think the more common scenario might be to increase the bucket size instead of decreasing the request number.

italorossi replied:
Let's keep the behavior consistent with the function name and make this only increase.

victor-torres commented:
Remember to write the new test cases 😉

jamesmishra commented:
Anything I should change?
